core/storage/trie/impl/persistent_trie_batch_impl.hpp

```diff
@@ -16,18 +17,23 @@ namespace kagome::storage::trie {
   class PersistentTrieBatchImpl : public PersistentTrieBatch {
    public:
     using RootChangedEventHandler = std::function<void(const common::Buffer &)>;
+    enum class Error {
+      ASYNC_OPERATION_FAILED = 1,
+    };
 
     PersistentTrieBatchImpl(
         std::shared_ptr<Codec> codec,
         std::shared_ptr<TrieSerializer> serializer,
         boost::optional<std::shared_ptr<changes_trie::ChangesTracker>> changes,
-        std::unique_ptr<PolkadotTrie> trie,
+        std::shared_ptr<PolkadotTrie> trie,
         RootChangedEventHandler &&handler);
     ~PersistentTrieBatchImpl() override = default;
 
     outcome::result<Buffer> commit() override;
     std::unique_ptr<TopperTrieBatch> batchOnTop() override;
 
+    void init();
```

Comment:
Better add a static factory method and make the constructor private, otherwise it's very easy to forget to call init().

Reply:
agree